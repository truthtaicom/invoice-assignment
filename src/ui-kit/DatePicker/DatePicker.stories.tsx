import { storiesOf } from '@storybook/react';
import React from 'react';
import { action } from '@storybook/addon-actions';
import DatePicker from "./DatePicker"

storiesOf("UI-Kit/DatePicker", module)
  .add("basic", () => (
    <DatePicker
      selected={new Date()}
      onChange={action('onChange')}
      dateFormat="yyyy-MM-dd"
    />
  ))
  .add("with label", () => (
    <DatePicker
      label="Date"
      selected={new Date()}
      onChange={action('onChange')}
      dateFormat="yyyy-MM-dd"
    />
  ))