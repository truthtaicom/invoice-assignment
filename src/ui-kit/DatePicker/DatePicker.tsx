import React from 'react'
import styled from 'styled-components'
import ReactDatePicker, { ReactDatePickerProps } from "react-datepicker";
import * as Colors from '../Variables/Colors'
import { Input } from '../'
import "react-datepicker/dist/react-datepicker.css";

export interface IDatePicker {
  label?: string
}

const DatePickerWrapper = styled.div`
  input {
    display: block;
    width: calc(100% - 1.75rem);
    padding: .375rem .75rem;
    font-size: 1rem;
    line-height: 1.5;
    color: #495057;
    background-color: ${Colors.white};
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: .25rem;
    transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  }
`


const DatePicker: React.FC<ReactDatePickerProps & any> = (props) => {
  return (
    <DatePickerWrapper>
      <Input {...props}>
        <ReactDatePicker {...props} selected={props.selectedDate} />
      </Input>
    </DatePickerWrapper>
  )
}

export default DatePicker;