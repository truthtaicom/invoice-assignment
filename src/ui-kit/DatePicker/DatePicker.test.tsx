import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import DatePicker from './DatePicker'

describe('<DatePicker /> spec', () => {
  it('renders the component', () => {
    const component = render(<DatePicker selectedDate={new Date()} />)
    expect(component).toMatchSnapshot()
  })
});