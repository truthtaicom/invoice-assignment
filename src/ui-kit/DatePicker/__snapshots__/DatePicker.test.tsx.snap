// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-kEYyzF gVtqmL"
      >
        <div
          class="sc-bwzfXH jqqcYM"
        >
          <div
            class="sc-htpNat caMxVp"
          >
            <span
              class="sc-bZQynM hanFHq"
            />
            <span
              class="sc-bZQynM hanFHq"
            >
              2019-05-02
            </span>
            <span
              class="sc-bZQynM hanFHq"
            >
              ^
            </span>
          </div>
          <div
            class="sc-htpNat eWpxpu"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  type="text"
                  value="05/02/2019"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kEYyzF gVtqmL"
    >
      <div
        class="sc-bwzfXH jqqcYM"
      >
        <div
          class="sc-htpNat caMxVp"
        >
          <span
            class="sc-bZQynM hanFHq"
          />
          <span
            class="sc-bZQynM hanFHq"
          >
            2019-05-02
          </span>
          <span
            class="sc-bZQynM hanFHq"
          >
            ^
          </span>
        </div>
        <div
          class="sc-htpNat eWpxpu"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class=""
                type="text"
                value="05/02/2019"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
